src/main/java/oop/animals/enclosures/EnclosureClone.java

```diff
@@ -0,0 +1,26 @@
+package oop.animals.enclosures;
+
+import oop.animals.model.mammals.artiodactyls.Cow;
+
+public class EnclosureClone {
+
+    public Enclosure<Cow> cloneEnclosureOfCows(Enclosure<Cow> enclosureToClone) throws CloneNotSupportedException {
+        Enclosure<Cow> clonedEnclosure = new Enclosure<>();
+        Cow clonedCow;
+        for (Cow cow : enclosureToClone.getAnimals()) {
+            clonedCow = cow.clone();
+            clonedEnclosure.addAnimal(clonedCow);
+        }
+        return clonedEnclosure;
+    }
+
+    @Override
+    public int hashCode() {
```

Comment:
This is a service, you don't need to override `hashCode` and `equals` methods. Moreover, they are made incorrectly in your implementation.